Updated esXai contract to include bulk read functions for redemptions

XDapps opened the pull request:
[Ticket](https://www.pivotaltracker.com/story/show/188462806)

Updated esXai2 contract to include two optimized reader functions for redemptions. 

Testing, tested by deploying on Sepolia and confirming the read functions work through the explorer.

CryptITAustria reviewed (approved):
@XDapps I added the update for the upgrade esXai script, please have a look.

I tested this by redeploying esXai2 and then esXai3 to sepolia.

I fixed a bug in the esXai3 deploy script there too both tested on sepolia.

Additionally I added the new functions to esXai3 too since they **NEED** to be there in any case.

XDapps commented:
> @XDapps I added the update for the upgrade esXai script, please have a look.
> 
> I tested this by redeploying esXai2 and then esXai3 to sepolia.
> 
> I fixed a bug in the esXai3 deploy script there too both tested on sepolia.
> 
> Additionally I added the new functions to esXai3 too since they **NEED** to be there in any case.

@CryptITAustria looks good to me, thank you!
